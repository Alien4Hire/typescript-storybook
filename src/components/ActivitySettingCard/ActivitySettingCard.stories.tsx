import React from 'react';
import { useArgs } from '@storybook/client-api';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import ProviderWrapper from '../../util/ProviderWrapper';
import CardElement from './index';

export default {
  title: 'Component/ActivitySettingCard',
  component: CardElement,
  decorators: [
    (Story) => (
      <ProviderWrapper>
        <Story />
      </ProviderWrapper>
    ),
  ],
} as ComponentMeta<typeof CardElement>;

const Template: ComponentStory<typeof CardElement> = (args) => {
  const [, updateArgs] = useArgs();

  const onChecked = (
    e: React.ChangeEvent<HTMLInputElement>,
    checked: boolean
  ) => {
    updateArgs({ checked: true });
    args.onChecked?.(e, checked);
  };

  return <CardElement {...args} onChecked={onChecked} />;
};

export const ActivitySettingCard = Template.bind({});
ActivitySettingCard.args = {
  cardInfo: {
    id: '2',
    title: 'Example activity settings rule',
    gradingPolicy: 'Test/Quiz',
    recommendedUse: 'Post-Class',
    visibility: 'Available',
  },
  checked: false,
};
