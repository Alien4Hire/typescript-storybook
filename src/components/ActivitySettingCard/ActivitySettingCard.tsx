import React from 'react';
import {
  Button,
  FormControlLabel,
  Grid,
  Radio,
  Typography,
} from '@mui/material';
import { SwitchBaseProps } from '@mui/material/internal/SwitchBase';
import {
  InfoWrapper,
  StyledCard,
  TitleWrapper,
} from './ActivitySettingCardStyles';
import { theme } from '../../theme';

import { ReactComponent as TrashIcon } from '../../icons/altTrash.svg';

export type ActivitySettingCardT = {
  id: string;
  title: string;
  gradingPolicy: string;
  recommendedUse: string;
  visibility: string;
};

export type CardPropsT = {
  cardInfo: ActivitySettingCardT;
  checked: boolean;
  onChecked: SwitchBaseProps['onChange'];
  onDelete: (id: string) => void;
};

type RadioNativePropsT = React.InputHTMLAttributes<HTMLInputElement> & {
  'data-testid'?: string;
};
const ActivitySettingCard = ({
  cardInfo,
  checked,
  onChecked,
  onDelete,
}: CardPropsT): JSX.Element => {
  const handleDelete = () => onDelete(cardInfo.id);

  return (
    <StyledCard theme={theme}>
      <Grid container alignItems="center" padding="20px">
        <Grid item xs={5}>
          <TitleWrapper>
            <FormControlLabel
              control={
                <Radio
                  value={cardInfo.id}
                  checked={checked}
                  onChange={onChecked}
                  inputProps={
                    { 'data-testid': 'RadioButton' } as RadioNativePropsT
                  }
                />
              }
              label={<Typography variant="h3">{cardInfo.title}</Typography>}
            />
          </TitleWrapper>
        </Grid>
        <Grid item xs={2.5}>
          <InfoWrapper maxWidth="135px">
            <Typography
              sx={{
                color: theme.palette.gray.veryDark,
                lineHeight: '24px',
              }}
              variant="textBasic"
            >
              Grading Policy:
            </Typography>
            <Typography sx={{ fontWeight: 'bold' }}>
              {cardInfo.gradingPolicy}
            </Typography>
          </InfoWrapper>
        </Grid>
        <Grid item xs={2.5}>
          <InfoWrapper>
            <Typography
              variant="textBasic"
              sx={{
                color: theme.palette.gray.veryDark,
                lineHeight: '24px',
              }}
            >
              Recommend Use:
            </Typography>
            <Typography variant="h4">{cardInfo.recommendedUse}</Typography>
          </InfoWrapper>
        </Grid>
        <Grid item xs={1.5}>
          <InfoWrapper>
            <Typography
              variant="textBasic"
              sx={{
                color: theme.palette.gray.veryDark,
                lineHeight: '24px',
              }}
            >
              Visibility:
            </Typography>
            <Typography variant="h4">{cardInfo.visibility}</Typography>
          </InfoWrapper>
        </Grid>
        <Grid item xs={0.5}>
          <Button
            startIcon={<TrashIcon />}
            data-testid="IconButton"
            onClick={handleDelete}
          />
        </Grid>
      </Grid>
    </StyledCard>
  );
};

export default ActivitySettingCard;
