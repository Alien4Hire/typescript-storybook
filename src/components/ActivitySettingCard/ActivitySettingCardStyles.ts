import { Card, styled } from '@mui/material';
import Box from '@mui/material/Box';

export const StyledCard = styled(Card)(({ theme }) => ({
  border: `1px solid ${theme.palette.gray.medium}`,
  boxShadow: 'none',
}));

export const TitleWrapper = styled(Box)(() => ({
  display: 'flex',
  alignItems: 'center',
}));

export const InfoWrapper = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
});
