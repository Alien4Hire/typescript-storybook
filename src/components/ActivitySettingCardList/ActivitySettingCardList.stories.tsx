import React from 'react';
import { useArgs } from '@storybook/client-api';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import ProviderWrapper from '../../util/ProviderWrapper';
import { default as ListElement, mockCardsList } from './index';

export default {
  title: 'Component/ActivitySettingCardList',
  component: ListElement,
  decorators: [
    (Story) => (
      <ProviderWrapper>
        <Story />
      </ProviderWrapper>
    ),
  ],
} as ComponentMeta<typeof ListElement>;

const Template: ComponentStory<typeof ListElement> = (args) => {
  const [, updateArgs] = useArgs();

  const onChecked = (
    e: React.ChangeEvent<HTMLInputElement>,
    checked: boolean
  ) => {
    updateArgs({ checkedId: e.target.value });
    args.onChecked?.(e, checked);
  };

  const onDelete = (id: string) => {
    const newCards = args.cards.filter((card) => card.id !== id);
    updateArgs({ cards: newCards });
    args.onDelete?.(id);
  };

  return <ListElement {...args} onChecked={onChecked} onDelete={onDelete} />;
};

export const ActivitySettingCardList = Template.bind({});
ActivitySettingCardList.args = {
  cards: mockCardsList,
  checkedId: '1',
};
