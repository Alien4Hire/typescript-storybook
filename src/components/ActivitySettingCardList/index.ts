export { default } from './ActivitySettingCardList';

export const mockCard = {
  title: 'Example activity settings rule',
  gradingPolicy: 'Test/Quiz',
  recommendedUse: 'Post-Class',
  visibility: 'Available',
} as const;

export const mockCardsList = [...Array(5)].map((_, index) => ({
  ...mockCard,
  id: String(index),
  title: mockCard.title + ' #' + (index + 1),
}));
