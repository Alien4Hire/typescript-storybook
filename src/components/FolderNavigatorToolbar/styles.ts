import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
