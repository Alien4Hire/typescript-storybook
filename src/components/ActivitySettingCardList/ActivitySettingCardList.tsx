import { Grid } from '@mui/material';
import { SwitchBaseProps } from '@mui/material/internal/SwitchBase';
import ActivitySettingCard from '../ActivitySettingCard';
import { ActivitySettingCardT } from '../ActivitySettingCard/ActivitySettingCard';

export type CardListPropsT = {
  checkedId: string;
  cards: ActivitySettingCardT[];
  onChecked: SwitchBaseProps['onChange'];
  onDelete: (id: string) => void;
};

const CardList = ({
  cards,
  checkedId,
  onChecked,
  onDelete,
}: CardListPropsT): JSX.Element => {
  return (
    <Grid container flexDirection="column" spacing={2}>
      {cards.map((card) => (
        <Grid item key={card.id}>
          <ActivitySettingCard
            cardInfo={card}
            checked={checkedId === card.id}
            onChecked={onChecked}
            onDelete={onDelete}
          />
        </Grid>
      ))}
    </Grid>
  );
};

export default CardList;
