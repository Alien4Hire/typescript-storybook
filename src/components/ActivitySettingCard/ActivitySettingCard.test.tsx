import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import renderer from 'react-test-renderer';

import { theme } from '../../theme';
import '../../setupTests';

import ActivitySettingCard from './index';
import { mockCard } from '../ActivitySettingCardList';
import { CardPropsT } from './ActivitySettingCard';

beforeAll(() => {
  window.getComputedStyle = jest.fn();
});

const baseProps = {
  cardInfo: { ...mockCard, id: '1' },
  checked: false,
  onChecked: jest.fn(),
  onDelete: jest.fn(),
};

const Component = (props: CardPropsT) => (
  <StyledEngineProvider injectFirst>
    <ThemeProvider theme={theme}>
      <ActivitySettingCard {...props} />
    </ThemeProvider>
  </StyledEngineProvider>
);

const renderComponent = (customProps: Partial<CardPropsT> = {}) => {
  const props = { ...baseProps, ...customProps };

  return render(<Component {...props} />);
};

describe('<ViewCard />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Component {...baseProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correct content', () => {
    renderComponent();

    const { id, ...fieldsForCheck } = baseProps.cardInfo;
    for (const label of Object.values(fieldsForCheck)) {
      expect(screen.getByText(label)).toBeInTheDocument();
    }
  });

  it('should render with checked radio button', () => {
    renderComponent({ checked: true });
    const radioBtn: HTMLInputElement = screen.getByTestId('RadioButton');
    expect(radioBtn.checked).toBe(true);
    expect(radioBtn.value).toBe(baseProps.cardInfo.id);
  });

  it('should call onChecked function', () => {
    renderComponent();

    const radioBtn = screen.getByTestId('RadioButton');
    expect(baseProps.onChecked).not.toBeCalled();
    fireEvent.click(radioBtn);

    expect(baseProps.onChecked).toBeCalledTimes(1);
  });

  it('should call onDelete function', () => {
    renderComponent();

    const deleteBtn = screen.getByTestId('IconButton');
    expect(baseProps.onDelete).not.toBeCalled();
    fireEvent.click(deleteBtn);

    expect(baseProps.onDelete).toBeCalledTimes(1);
  });
});
