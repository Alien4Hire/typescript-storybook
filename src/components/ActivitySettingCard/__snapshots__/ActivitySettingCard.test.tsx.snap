// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewCard /> renders correctly 1`] = `
.c12 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c0 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7::-moz-focus-inner {
  border-style: none;
}

.c7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c15 {
  margin: 0;
  color: #080808;
  font-size: 18px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.167;
}

.c19 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 400;
  font-style: normal;
  color: #383838;
  line-height: 24px;
}

.c20 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  font-weight: 700;
}

.c21 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.235;
}

.c17 {
  max-width: 135px;
}

.c24 {
  font-size: 14px;
  text-transform: uppercase;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #00758e;
}

.c24:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0,117,142,0.04);
}

.c24.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c25 {
  display: inherit;
  margin-right: 8px;
  margin-left: -4px;
}

.c25 > *:nth-of-type(1) {
  font-size: 20px;
}

.c1 {
  overflow: hidden;
}

.c8 {
  padding: 9px;
  border-radius: 50%;
}

.c10 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  margin-left: -11px;
  margin-right: 16px;
}

.c6.Mui-disabled {
  cursor: default;
}

.c6 .MuiFormControlLabel-label.Mui-disabled {
  color: rgba(0,0,0,0.38);
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 41.666667%;
  -ms-flex-preferred-size: 41.666667%;
  flex-basis: 41.666667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 41.666667%;
}

.c16 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 20.833333%;
  -ms-flex-preferred-size: 20.833333%;
  flex-basis: 20.833333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 20.833333%;
}

.c22 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 12.5%;
  -ms-flex-preferred-size: 12.5%;
  flex-basis: 12.5%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 12.5%;
}

.c23 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 4.166667%;
  -ms-flex-preferred-size: 4.166667%;
  flex-basis: 4.166667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 4.166667%;
}

.c11 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c13 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c14 {
  left: 0;
  position: absolute;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition: -webkit-transform 150ms cubic-bezier(0.4,0,1,1) 0ms;
  -webkit-transition: transform 150ms cubic-bezier(0.4,0,1,1) 0ms;
  transition: transform 150ms cubic-bezier(0.4,0,1,1) 0ms;
}

.c9 {
  color: rgba(0,0,0,0.6);
}

.c9:hover {
  background-color: rgba(0,117,142,0.04);
}

.c9.Mui-checked {
  color: #00758e;
}

.c9.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c2 {
  border: 1px solid #c8c8c8;
  box-shadow: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media print {
  .c7 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c24:hover {
    background-color: transparent;
  }
}

@media (min-width:660px) {
  .c4 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

@media (min-width:768px) {
  .c4 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

@media (min-width:992px) {
  .c4 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

@media (min-width:1200px) {
  .c4 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

@media (min-width:660px) {
  .c16 {
    -webkit-flex-basis: 20.833333%;
    -ms-flex-preferred-size: 20.833333%;
    flex-basis: 20.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 20.833333%;
  }
}

@media (min-width:768px) {
  .c16 {
    -webkit-flex-basis: 20.833333%;
    -ms-flex-preferred-size: 20.833333%;
    flex-basis: 20.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 20.833333%;
  }
}

@media (min-width:992px) {
  .c16 {
    -webkit-flex-basis: 20.833333%;
    -ms-flex-preferred-size: 20.833333%;
    flex-basis: 20.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 20.833333%;
  }
}

@media (min-width:1200px) {
  .c16 {
    -webkit-flex-basis: 20.833333%;
    -ms-flex-preferred-size: 20.833333%;
    flex-basis: 20.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 20.833333%;
  }
}

@media (min-width:660px) {
  .c22 {
    -webkit-flex-basis: 12.5%;
    -ms-flex-preferred-size: 12.5%;
    flex-basis: 12.5%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 12.5%;
  }
}

@media (min-width:768px) {
  .c22 {
    -webkit-flex-basis: 12.5%;
    -ms-flex-preferred-size: 12.5%;
    flex-basis: 12.5%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 12.5%;
  }
}

@media (min-width:992px) {
  .c22 {
    -webkit-flex-basis: 12.5%;
    -ms-flex-preferred-size: 12.5%;
    flex-basis: 12.5%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 12.5%;
  }
}

@media (min-width:1200px) {
  .c22 {
    -webkit-flex-basis: 12.5%;
    -ms-flex-preferred-size: 12.5%;
    flex-basis: 12.5%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 12.5%;
  }
}

@media (min-width:660px) {
  .c23 {
    -webkit-flex-basis: 4.166667%;
    -ms-flex-preferred-size: 4.166667%;
    flex-basis: 4.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 4.166667%;
  }
}

@media (min-width:768px) {
  .c23 {
    -webkit-flex-basis: 4.166667%;
    -ms-flex-preferred-size: 4.166667%;
    flex-basis: 4.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 4.166667%;
  }
}

@media (min-width:992px) {
  .c23 {
    -webkit-flex-basis: 4.166667%;
    -ms-flex-preferred-size: 4.166667%;
    flex-basis: 4.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 4.166667%;
  }
}

@media (min-width:1200px) {
  .c23 {
    -webkit-flex-basis: 4.166667%;
    -ms-flex-preferred-size: 4.166667%;
    flex-basis: 4.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 4.166667%;
  }
}

@media (hover:none) {
  .c9:hover {
    background-color: transparent;
  }
}

<div
  className="c0 MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c1 MuiCard-root c2"
>
  <div
    className="c3 MuiGrid-root MuiGrid-container"
  >
    <div
      className="c4 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
    >
      <div
        className="c5 MuiBox-root"
      >
        <label
          className="c6 MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd"
        >
          <span
            className="c7 MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root c8 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root c9"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={false}
              className="c10 PrivateSwitchBase-input"
              data-testid="RadioButton"
              onChange={[Function]}
              type="radio"
              value="1"
            />
            <span
              className="c11"
            >
              <svg
                aria-hidden={true}
                className="c12 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c13"
                data-testid="RadioButtonUncheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden={true}
                className="c12 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c14"
                data-testid="RadioButtonCheckedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </span>
          </span>
          <h3
            className="c15 MuiTypography-root MuiTypography-h3"
          >
            Example activity settings rule
          </h3>
        </label>
      </div>
    </div>
    <div
      className="c16 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2.5"
    >
      <div
        className="c17 c18 MuiBox-root"
      >
        <span
          className="c19 MuiTypography-root MuiTypography-textBasic"
        >
          Grading Policy:
        </span>
        <p
          className="c20 MuiTypography-root MuiTypography-body1"
        >
          Test/Quiz
        </p>
      </div>
    </div>
    <div
      className="c16 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2.5"
    >
      <div
        className="c18 MuiBox-root"
      >
        <span
          className="c19 MuiTypography-root MuiTypography-textBasic"
        >
          Recommend Use:
        </span>
        <h4
          className="c21 MuiTypography-root MuiTypography-h4"
        >
          Post-Class
        </h4>
      </div>
    </div>
    <div
      className="c22 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1.5"
    >
      <div
        className="c18 MuiBox-root"
      >
        <span
          className="c19 MuiTypography-root MuiTypography-textBasic"
        >
          Visibility:
        </span>
        <h4
          className="c21 MuiTypography-root MuiTypography-h4"
        >
          Available
        </h4>
      </div>
    </div>
    <div
      className="c23 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-0.5"
    >
      <button
        className="c7 MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root c24"
        data-testid="IconButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="c25 MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg>
            altTrash.svg
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
